packages/amplify-category-auth/provider-utils/awscloudformation/triggers/CreateAuthChallenge/function-template-dir/trigger-index.js

```diff
@@ -1,14 +1,58 @@
-/*
-  this file will loop through all js modules which are uploaded to the lambda resource,
-  provided that the file names (without extension) are included in the "MODULES" env variable.
-  "MODULES" is a comma-delimmited string.
-*/
+/**
+ * @fileoverview
+ *
+ * This CloudFormation Trigger creates a Lambda function which waits for all
+ * other specified Lambdas (modules) to resolve, which should be located in the
+ * same directory as this file (./).
+ *
+ * IMPORTANT: This handler *must* return the `event` since it may be consumed by
+ * additional layers in a CloudFormation stack. Failing to return an acceptable
+ * event may cause errors like "Invalid JSON" and prevent properly configured
+ * projects from deploying as expected.
+ *
+ * @see https://github.com/aws-amplify/amplify-cli/pull/7219
+ */
+
+/**
+ * The names of modules to load are stored as a comma-delimited string in the
+ * `MODULES` env variable.
+ */
 const moduleNames = process.env.MODULES.split(',');
+/**
+ * The array of imported modules.
+ */
 const modules = moduleNames.map(name => require(`./${name}`));
 
-exports.handler = (event, context, callback) => {
-  for (let i = 0; i < modules.length; i += 1) {
-    const { handler } = modules[i];
-    handler(event, context, callback);
-  }
+/**
+ * This handler is itself a Lambda function which iterates over all of the given
+ * modules and awaits them.
+ *
+ * @see https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html
+ *
+ * @param {object} event
+ *
+ * The event that triggered this Lambda.
+ *
+ * @param {object} context
+ *
+ * The context for this Lambda. See:
+ * https://docs.aws.amazon.com/lambda/latest/dg/nodejs-context.html
+ *
+ * @param {function} callback
+ *
+ * A deprecated way to send a response synchronously. See:
+ * https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html#nodejs-handler-sync
+ *
+ * @returns
+ *
+ * The CloudFormation trigger event.
+ */
+exports.handler = async (event, context, callback) => {
+  /**
+   * Instead of naively iterating over all handlers, run them concurrently with
+   * `await Promise.all(...)`. This would otherwise just be determined by the
+   * order of names in the `MODULES` var.
+   */
```

Comment:
I think this comment can be removed as well. The legacy implementation would behave the same way if it was calling asynchronous code.

Reply:
This is just clarifying that instead of `for ... of`, we're using `await Promise.all(...)`. I would leave it in, but I will defer to your judgment.

Reply:
See:

https://github.com/aws-amplify/amplify-cli/blob/63e647bb257b8bee4a5bdc12d6902b1375a4da12/packages/amplify-category-auth/provider-utils/awscloudformation/triggers/VerifyAuthChallengeResponse/captcha-verify.js#L10-L21